Create Monero tutorials (#95)
PurpleI2P/i2pd_docs_en: docs/tutorials/monero.md

Monero
==================

## Connect to remote nodes.

To connect Monero nodes, you have to configure the wallet software:

In [Feather wallet](http://rwzulgcql2y3n6os2jhmhg6un2m33rylazfnzhf56likav47aylq.b32.i2p/): File -> Settings -> Network -> Proxy > i2p, with this option all network traffic will be routed over i2p, including websocket connection (if enabled).

## Connection between other I2P nodes.

1) Configure i2pd to create server type tunnel and put in your `./tunnels.conf` file:

    [monero-node]
    type = server
    host = 127.0.0.1

>     port = 18080 # P2P port

Reviewer: ```suggestion
    port = 18085 # P2P port
```

Author: why different port needed?

Author: why different port used?

Reviewer: 18080 is the p2p port. Anonymous inbound uses custom/new ports

    keys = monero-node.dat

2) Restart i2pd or send `SIGHUP` signal.

3) Find your b32 address of the node.

>    Go to webconsole -> [I2P tunnels page](http://127.0.0.1:7070/?page=i2p_tunnels). Look for Server tunnels and you will see address like \<long random string\>.b32.i2p next to monero-node.

Reviewer: ```suggestion
   You can view the RPC i2p.b32 address with:
`curl http://127.0.0.1:7070/?page=i2p_tunnels 2>&1 | grep -Eo "[a-zA-Z0-9./?=_%:-]*" | grep "18085"`
```

   The node is now available in Invisible Internet. You can check the node's status by visiting \<long random string\>.b32.i2p/get_info.

4) Update your monero.conf file;

>     anonymous-inbound=XXX.b32.i2p:18080,127.0.0.1:18080

Reviewer: ```suggestion
    anonymous-inbound=XXX.b32.i2p,127.0.0.1:18085
```

Replace XXX with your b32 address.

## Setup I2P service on already running public node.

Open wallet inteface (the "RPC") allows anyone to connect their wallets to Monero network through the node. This is useful for users who don't run their own nodes and connecting over i2p. 

If you already have a node with public RPC and now want to run i2p service, follow those steps:

1) Make sure the node is already configured for public use and running. (For an example you setted up `monero.conf` such as `public-node=1` with `restricted-rpc=1`)

2) Configure i2pd to create server type tunnel and put in your `./tunnels.conf` file:

>     [monero-node-rpc]

Reviewer: ```suggestion
    [monero-rpc]
```

    type = server
    host = 127.0.0.1
    port = 18089 # Restricted RPC port

>     keys = monero-node.dat

Reviewer: ```suggestion
    keys = monero-rpc.dat
```

Author: i think its better to keep one key file to have one address instead separate addresses for RPC and P2P

Reviewer: For privacy, i personally think its better to have rpc and the node uncorrelated

you _can_ use the same key for both, but imo you shouldn't share your p2p address

3) Restart i2pd or send `SIGHUP` signal.

> 4) Find your b32 address of the node.

Reviewer: ```suggestion
4) Find your b32 address of the node's rpc.
```

>    Go to webconsole -> [I2P tunnels page](http://127.0.0.1:7070/?page=i2p_tunnels). Look for Server tunnels and you will see address like \<long random string\>.b32.i2p next to monero-node.

Reviewer: ```suggestion
   You can view the RPC i2p.b32 address with:
`curl http://127.0.0.1:7070/?page=i2p_tunnels 2>&1 | grep -Eo "[a-zA-Z0-9./?=_%:-]*" | grep "18089"`
```

Author: Can be confusing for beginners.

Reviewer: Its more confusing if you don't have a desktop environment. 

can add both instructions though?

Author: @nahuhh doesn't work, i2pd webconsole doesn't show the ports

Reviewer: 👀👀 it does on my system. Weird. Will check for  another method

In [Feather wallet](http://rwzulgcql2y3n6os2jhmhg6un2m33rylazfnzhf56likav47aylq.b32.i2p/): File -> Settings -> Network -> Proxy > i2p, with this option all network traffic will be routed over i2p, including websocket connection (if enabled).

## Connection between other I2P nodes.

Warning: The usage of I2P network is still considered experimental;

Sync the blockchain over I2P isn't supported and there are a few pessimistic cases where privacy is leaked.

The design is intended to maximize privacy of the source of a transaction by broadcasting it over an anonymity network, while relying on clearnet for the remainder of messages to make surrounding node attacks (via sybil) more difficult. 

1) Configure i2pd to create server type tunnel and put in your `./tunnels.conf` file:

    [monero-node]
    type = server
    host = 127.0.0.1
    port = 18085 # Anonymous inbound port
    keys = monero-node.dat

2) Restart i2pd or send `SIGHUP` signal.

3) Find your b32 address of the node.

   Go to webconsole -> [I2P tunnels page](http://127.0.0.1:7070/?page=i2p_tunnels). Look for Server tunnels and you will see address like \<long random string\>.b32.i2p next to monero-node.

>    Or get with curl: `curl http://127.0.0.1:7070/?page=i2p_tunnels 2>&1 | grep -Eo "[a-zA-Z0-9./?=_%:-]*" | grep "18089"`

Reviewer: ```suggestion
   Or get with curl: `curl http://127.0.0.1:7070/?page=i2p_tunnels 2>&1 | grep -Eo "[a-zA-Z0-9./?=_%:-]*" | grep "18085"`
```

>    The node is now available in Invisible Internet. You can check the node's status by visiting \<long random string\>.b32.i2p/get_info.

Reviewer: Can drop this line (rpc isnt setup yet)

4) Update your monero.conf file;

>     anonymous-inbound=XXX.b32.i2p:18085,127.0.0.1:18085

Reviewer: ```suggestion
    anonymous-inbound=XXX.b32.i2p,127.0.0.1:18085
```

no port on the b32

Author: if the user uses same key for both p2p and RPC then the port would be useful

Reviewer: Monerod doesnt accept a port (uses port 0, aka the lowest port on the address)

    tx-proxy=i2p,127.0.0.1:4447,disable_noise

Replace XXX with your b32 address.

## Setup I2P service on already running public node.

Open wallet inteface (the "RPC") allows anyone to connect their wallets to Monero network through the node. This is useful for users who don't run their own nodes and connecting over i2p. 

If you already have a node with public RPC and now want to run i2p service, follow those steps:

> 1) Make sure the node is already configured for public use and running. (For an example you setted up `monero.conf` such as `public-node=1` with `restricted-rpc=1`)

Reviewer: ```suggestion
1) Make sure the node is running and configured with a restricted rpc port by adding `rpc-restricted-bind-port=18089` to your monerod startup flags or config file.
```
